### PR TITLE
Patch for NMT accuracy regression from PTL 1.1.3 to PTL 1.1.4+

### PR DESCRIPTION
**Pull request by ericharper**
The default for `find_unused_parameters` was changed from `True` to `False` in PTL. We're seeing higher accuracy in NMT with `find_unused_parameters=True`. Overriding this in NMT model until we learn more.

### CODE REVIEW

File: nemo/core/classes/modelPT.py [hunk 1]
```diff
@@ -1206,6 +1208,16 @@ def _update_dataset_config(self, dataset_name: str, config: Optional[Union[DictC
             else:
                 raise ValueError("`dataset_name` when updating config must be one of [train, validation, test]")
 
+    # def configure_ddp(self, model: LightningModule, device_ids: List[int]) -> DistributedDataParallel:
```
Comment (titu1994) [inline nemo/core/classes/modelPT.py:1211]: Why is this not a core level fix ?

Comment (ericharper) [reply nemo/core/classes/modelPT.py:1211]: It was failing unit tests.

Comment (titu1994) [reply nemo/core/classes/modelPT.py:1211]: Hmm wasn't it the default in 1.1.3 anyway ? It's odd that unit tests should fail

File: requirements/requirements.txt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 numpy>=1.18.2
 onnx>=1.7.0
-pytorch-lightning>=1.1.0,<=1.1.3
+pytorch-lightning>=1.1.0,<=1.1.5
```
Comment (okuchaiev) [inline requirements/requirements.txt:3]: do we need the <=1.1.5 part?

File: nemo/collections/nlp/modules/common/transformer/transformer_generators.py [hunk 3]
```diff
@@ -188,7 +190,7 @@ def __init__(self, embedding, decoder, log_softmax, beam_size=1, temperature=1.0
         self.beam_size = beam_size
         self.temp = temperature
 
-    @torch.no_grad()
+    # @torch.no_grad()
```
Comment (okuchaiev) [inline nemo/collections/nlp/modules/common/transformer/transformer_generators.py:193]: just wondering - why these no_grad() are no longer needed?

File: requirements/requirements.txt (hunk 2 above)
Comment (ericharper) [reply requirements/requirements.txt:3]: 1.1.6 was breaking MelGan.

File: nemo/collections/nlp/modules/common/transformer/transformer_generators.py (hunk 3 above)
Comment (ericharper) [reply nemo/collections/nlp/modules/common/transformer/transformer_generators.py:193]: I was thinking that weights should be frozen anyway during eval, so that these calls would be redundant.

File: nemo/collections/nlp/models/machine_translation/mt_enc_dec_config.py [hunk 4]
```diff
@@ -34,6 +34,7 @@ class MTEncDecModelConfig(EncDecNLPModelConfig):
     len_pen: float = 0.0
     max_generation_delta: int = 3
     label_smoothing: Optional[float] = 0.0
+    find_unused_parameters: Optional[bool] = False
```
[the remark this answers is not in the document]
Comment (ericharper) [reply nemo/collections/nlp/models/machine_translation/mt_enc_dec_config.py:37]: Yes, the goal was to get the same accuracy with False.

File: nemo/collections/nlp/modules/common/transformer/transformer_generators.py (hunk 3 above)
Comment (ericharper) [reply nemo/collections/nlp/modules/common/transformer/transformer_generators.py:193]: PTL calls .eval() and .no_grad() for val and test: https://pytorch-lightning.readthedocs.io/en/stable/converting.html#find-the-val-loop-meat

Comment (titu1994) [reply nemo/collections/nlp/modules/common/transformer/transformer_generators.py:193]: What about using these classes outside of PTL context?

File: nemo/collections/nlp/models/machine_translation/mt_enc_dec_config.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (ericharper) [reply nemo/collections/nlp/models/machine_translation/mt_enc_dec_config.py]: Changed to True and accuracy with False is still low.

**Review by okuchaiev — APPROVED** [review]
lgtm, thanks!
